Ports/lite-xl/patches/0001-Add-SerenityOS-platform-support.patch

```diff
@@ -0,0 +1,144 @@
+Date: Mon, 3 Oct 2022 18:21:44 +0200
+Subject: [PATCH] Add SerenityOS platform support
+
+Signed-off-by: Jan200101 <sentrycraft123@gmail.com>
+---
+ meson.build                       |  6 +++-
+ src/api/dirmonitor/inodewatcher.c | 54 +++++++++++++++++++++++++++++++
+ src/main.c                        |  6 ++--
+ src/meson.build                   |  2 ++
+ 4 files changed, 65 insertions(+), 3 deletions(-)
+ create mode 100644 src/api/dirmonitor/inodewatcher.c
+
+diff --git a/meson.build b/meson.build
+index 9454fce..57c7722 100644
+--- a/meson.build
++++ b/meson.build
+@@ -82,13 +82,17 @@ if not get_option('source-only')
+ 
+     foreach lua : lua_names
+         last_lua = (lua == lua_names[-1] or get_option('wrap_mode') == 'forcefallback')
+-        lua_dep = dependency(lua, fallback: last_lua ? ['lua', 'lua_dep'] : [], required : last_lua,
++        lua_dep = dependency(lua, fallback: last_lua ? ['lua', 'lua_dep'] : [], required : false,
+             version: '>= 5.4',
+             default_options: default_fallback_options + ['default_library=static', 'line_editing=false', 'interpreter=false']
+         )
+         if lua_dep.found()
+             break
+         endif
++
++        if last_lua
++            lua_dep = cc.find_library('lua', required: true)
++        endif
+     endforeach
+ 
+     pcre2_dep = dependency('libpcre2-8', fallback: ['pcre2', 'libpcre2_8'],
+diff --git a/src/api/dirmonitor/inodewatcher.c b/src/api/dirmonitor/inodewatcher.c
+new file mode 100644
+index 0000000..8ca0b4f
+--- /dev/null
++++ b/src/api/dirmonitor/inodewatcher.c
+@@ -0,0 +1,54 @@
++#include <stdlib.h>
++#include <unistd.h>
++#include <fcntl.h>
++#include <poll.h>
++#include <string.h>
++#include <limits.h>
++
++// C version of InodeWatcherEvent
++struct [[gnu::packed]] C_InodeWatcherEvent {
++    int watch_descriptor;
++    uint32_t type;
++    size_t name_length;
++    // This is a VLA which is written during the read() from the descriptor.
++    char const name[];
++};
++
++struct dirmonitor_internal {
++  int fd;
++};
++
++
++struct dirmonitor_internal* init_dirmonitor() {
++  struct dirmonitor_internal* monitor = calloc(sizeof(struct dirmonitor_internal), 1);
++  monitor->fd = create_inode_watcher(0);
++  return monitor;
++}
++
++
++void deinit_dirmonitor(struct dirmonitor_internal* monitor) {
++  close(monitor->fd);
++}
++
++
++
++int get_changes_dirmonitor(struct dirmonitor_internal* monitor, char* buffer, int length) {
++  return read(monitor->fd, buffer, length);
++}
++
++
++int translate_changes_dirmonitor(struct dirmonitor_internal* monitor, char* buffer, int length, int (*change_callback)(int, const char*, void*), void* data) {
++  struct C_InodeWatcherEvent* event = (struct C_InodeWatcherEvent*)buffer;
++  change_callback(event->watch_descriptor, NULL, data);
++  return 0;
++}
++
++
++int add_dirmonitor(struct dirmonitor_internal* monitor, const char* path) {
++  return inode_watcher_add_watch(monitor->fd, path, strlen(path), 31);
```

Comment:
This is another place where having access to InodeWatcherEvent would be good.

Right now `31` is just a magic number of all the InodeWatcherEvent Types added up